plugins/net_plugin/net_plugin.cpp

```diff
@@ -790,8 +983,22 @@ namespace eosio {
       void set_connection_type( const string& peer_addr );
       bool is_transactions_only_connection()const { return connection_type == transactions_only; } // thread safe, atomic
       bool is_blocks_only_connection()const { return connection_type == blocks_only; }
-      bool is_transactions_connection() const { return connection_type != blocks_only; } // thread safe, atomic
-      bool is_blocks_connection() const { return connection_type != transactions_only; } // thread safe, atomic
+
+      //add more connection type
+      bool is_peers_only_connection() const { return connection_type == peers_only; }
+
+      bool is_all_connection() const { return connection_type == all; }
+      bool is_both_connection() const { return connection_type == both; }
+
+      bool is_transactions_connection() const { return connection_type == transactions_only || is_both_connection() || is_all_connection(); }
+      bool is_blocks_connection() const { return connection_type == blocks_only || is_both_connection() || is_all_connection(); }
+      bool is_peers_connection() const { return connection_type == peers_only || is_all_connection(); }
+
+      bool is_no_transactions_connection() const { return !is_transactions_connection(); }
```

Comment:
I would prefer using `!is_transactions_connection()` instead of adding `is_no_transactions_connection()`